pkg/resource/manager.go

```diff
@@ -125,12 +144,245 @@ func (m *manager) BatchCreateGenericResources(ctx context.Context, request *pb.B
+				itemError = fmt.Errorf("error creating generic resource [%d] %s: %s", item.Create.Status, item.Create.Error.Type, item.Create.Error.Reason)
+			} else if item.Index != nil && item.Index.Error != nil {
+				itemError = fmt.Errorf("error re-indexing generic resource [%d] %s: %s", item.Index.Status, item.Index.Error.Type, item.Index.Error.Reason)
+			}
+			if itemError != nil {
+				bulkItemErrors = append(bulkItemErrors, itemError)
+			}
+		}
+	}
+
+	if len(bulkItemErrors) > 0 {
+		return fmt.Errorf("failed to create/update some generic resource versions: %v", bulkItemErrors)
 	}
 
 	return nil
 }
 
-func genericResourceId(r *pb.GenericResource) string {
-	return fmt.Sprintf("%s:%s", r.Type, r.Name)
+func (m *manager) ListGenericResources(ctx context.Context, request *pb.ListGenericResourcesRequest) (*pb.ListGenericResourcesResponse, error) {
+	// generic resources and their versions are stored in the same index. we need to specify the join field
+	// in this query in order to only select generic resources. we use a "must" here so we can combine this query with
+	// the one generated by the filter, if provided
+	queries := filtering.Must{
+		&filtering.Query{
+			Term: &filtering.Term{
+				genericResourceDocumentJoinField: genericResourceRelationName,
+			},
+		},
+	}
+
+	if request.Filter != "" {
+		filterQuery, err := m.filterer.ParseExpression(request.Filter)
+		if err != nil {
+			return nil, err
+		}
+
+		queries = append(queries, filterQuery)
+	}
+
+	searchRequest := &esutil.SearchRequest{
+		Index: m.indexManager.AliasName(genericResourcesDocumentKind, ""),
+		Search: &esutil.EsSearch{
+			Query: &filtering.Query{
+				Bool: &filtering.Bool{
+					Must: &queries,
+				},
+			},
+		},
+	}
+
+	if request.PageSize != 0 {
+		searchRequest.Pagination = &esutil.SearchPaginationOptions{
+			Size:  int(request.PageSize),
+			Token: request.PageToken,
+		}
+	}
+
+	searchResponse, err := m.esClient.Search(ctx, searchRequest)
+	if err != nil {
+		return nil, err
+	}
+
+	var genericResources []*pb.GenericResource
+	for _, hit := range searchResponse.Hits.Hits {
+		var genericResource pb.GenericResource
+		err = protojson.UnmarshalOptions{
+			DiscardUnknown: true,
+		}.Unmarshal(hit.Source, &genericResource)
+		if err != nil {
+			return nil, err
+		}
+
+		genericResource.Id = hit.ID
+		genericResources = append(genericResources, &genericResource)
+	}
+
+	return &pb.ListGenericResourcesResponse{
+		GenericResources: genericResources,
+		NextPageToken:    searchResponse.NextPageToken,
+	}, nil
+}
+
+// ListGenericResourceVersions handles the main logic for fetching versions associated with a generic resource.
+func (m *manager) ListGenericResourceVersions(ctx context.Context, request *pb.ListGenericResourceVersionsRequest) (*pb.ListGenericResourceVersionsResponse, error) {
+	return nil, nil
+}
+
+func (m *manager) GetGenericResource(ctx context.Context, resourceId string) (*pb.GenericResource, error) {
```

Comment:
Doesn't look like there are any tests for this method

Reply:
Yeah, this method wasn't actually needed since I trimmed the scope for this PR.  It will be needed for the actual implementation of `ListGenericResourceVersions`, so I left it to add tests later.